docs/reference-guides/data/data-core-customize-widgets.md

```diff
@@ -78,6 +78,7 @@ _Parameters_
 -   _value_ `boolean|Object`: Whether the inserter should be opened (true) or closed (false). To specify an insertion point, use an object.
 -   _value.rootClientId_ `string`: The root client ID to insert at.
 -   _value.insertionIndex_ `number`: The index to insert at.
+-   _value.initialCategory_ `string`: The tab category to display first when the block editor inserter is opened. A category corresponds to one of the tab categories defined in packages/block-editor/src/components/inserter/tabs.js.
```

Comment:
Can you explain the need for this new API for both the actions and the inserter component? What's this about?

Reply:
Thanks for taking a look.

The value `initialCategory` determines which tab displays when the inserter is opened. 

At the moment, the initial tab is blocks.

For example, when `setInserterIsOpened` is called with a value of: `initialCategory: 'patterns'`, the patterns tab will open first. 

Another option is 'media'. 

If none of those tabs exist, it will default to blocks.

The requirement is to be able to control this behaviour from the block editor.

The value is accessed via the [private selector](https://github.com/WordPress/gutenberg/pull/57572/files#diff-a1cdb9f3f4ce98ee4ac08627daff3c4cb003766468d4bb3780fe0be22a0ad87eR61).

This seemed to me to be the most obvious place to add the property at the time, especially seeing as its related to opening the inserter, but I'm very aware doesn't mean it's an appropriate place/method 😄 

If you have suggestions I'm very keen to make this better!

Reply:
I don't know if I have great suggestions here, adding it as part of the "insertion point" state could work but 

 - It should probably not say "category" but "tab" right?
 - Can we make it "private" everywhere?
 - Can we avoid experimental/unstable prefixes?

Reply:
Thanks for the tips!

> It should probably not say "category" but "tab" right?

👍🏻 

> Can we make it "private" everywhere?

Do you mean a new private action?

As with `rootClientId` and `insertionIndex` ([getInsertionPoint](https://github.com/WordPress/gutenberg/blob/3dba5c9f37cf4700252cda58f5b818215b356625/packages/editor/src/store/private-selectors.js#L25-L25)), the only way to access the property value is through the private [selector](https://github.com/WordPress/gutenberg/pull/57572/files#diff-a1cdb9f3f4ce98ee4ac08627daff3c4cb003766468d4bb3780fe0be22a0ad87eR61) in `@wordpress/editor`.

The other object properties `rootClientId` and `insertionIndex` are already out there in the wild as part of the "public" API. 

The sidebar inserter state I assume should remain in the `@wordpress/editor` package (?)

I'll have to take a closer look at the public [setIsInserterOpened](https://github.com/WordPress/gutenberg/blob/3dba5c9f37cf4700252cda58f5b818215b356625/packages/editor/src/store/actions.js#L703-L703) action, which lives in the `@wordpress/editor` package. `setIsInserterOpened` appears to be [exported](https://github.com/WordPress/gutenberg/blob/3dba5c9f37cf4700252cda58f5b818215b356625/packages/editor/src/components/provider/use-block-editor-settings.js#L261-L261) in `useBlockEditorSettings` from `@wordpress/editor` presumably so it can be accessed by blocks and elsewhere, e.g., 

```js
// From a block
select( blockEditorStore ).getSettings().__experimentalSetIsInserterOpened( {
	rootClientId: clientId,
	insertionIndex: 0,
} )
```

> Can we avoid experimental/unstable prefixes?

👍🏻 

This PR introduces `__experimentalInitialCategory/Tab`. 

I'm also happy to create a new follow up PR that removes `__experimental*` for `__experimentalFilterValue` and `__experimentalInsertionIndex`.

Cheers!